exchanges/gateio/gateio_websocket.go

```diff
@@ -761,11 +730,144 @@ func singleSymbolChannel(name string) bool {
+			return fmt.Errorf("%w for `%s` only `1` is supported", subscription.ErrInvalidLevel, channel)
+		}
+	case futuresOrderbookUpdateChannel:
+		if a == asset.Futures {
+			if levels != 20 && levels != 50 && levels != 100 {
+				return fmt.Errorf("%w for `%s` only `20`, `50` or `100` is supported", subscription.ErrInvalidLevel, channel)
+			}
+			if levels != 20 {
+				if interval == kline.TwentyMilliseconds {
+					return fmt.Errorf("%w for `%s` only `20` levels supports `%s`", subscription.ErrInvalidInterval, channel, kline.TwentyMilliseconds)
+				}
+				if interval != kline.HundredMilliseconds {
+					return fmt.Errorf("%w for `%s` only `%s` is supported", subscription.ErrInvalidInterval, channel, kline.HundredMilliseconds)
+				}
+			} else if interval != kline.HundredMilliseconds && interval != kline.TwentyMilliseconds {
+				return fmt.Errorf("%w for `%s` only `%s` or `%s` is supported", subscription.ErrInvalidInterval, channel, kline.HundredMilliseconds, kline.TwentyMilliseconds)
+			}
+		} else {
+			if interval != kline.HundredMilliseconds && interval != kline.ThousandMilliseconds {
+				return fmt.Errorf("%w for `%s` only `%s` or `%s` is supported", subscription.ErrInvalidInterval, channel, kline.HundredMilliseconds, kline.ThousandMilliseconds)
+			}
+			if levels != 5 && levels != 10 && levels != 20 && levels != 50 && levels != 100 {
+				return fmt.Errorf("%w for `%s` only `5`, `10`, `20`, `50` or `100` is supported", subscription.ErrInvalidLevel, channel)
+			}
+		}
+	case optionsOrderbookTickerChannel:
+		if interval != 0 {
+			return fmt.Errorf("%w for `%s` only `0` is supported", subscription.ErrInvalidInterval, channel)
+		}
+		if levels != 1 {
+			return fmt.Errorf("%w for `%s` only `1` is supported", subscription.ErrInvalidLevel, channel)
+		}
+	case optionsOrderbookUpdateChannel:
+		if interval != kline.HundredMilliseconds && interval != kline.ThousandMilliseconds {
+			return fmt.Errorf("%w for `%s` only `%s` or `%s` is supported", subscription.ErrInvalidInterval, channel, kline.HundredMilliseconds, kline.ThousandMilliseconds)
+		}
+		if levels != 5 && levels != 10 && levels != 20 && levels != 50 {
+			return fmt.Errorf("%w for `%s` only `5`, `10`, `20` or `50` is supported", subscription.ErrInvalidLevel, channel)
+		}
+	case optionsOrderbookChannel:
+		if interval != kline.TwoHundredAndFiftyMilliseconds {
+			return fmt.Errorf("%w for `%s` only `%s` is supported", subscription.ErrInvalidInterval, channel, kline.TwoHundredAndFiftyMilliseconds)
+		}
+		if levels != 1 && levels != 5 && levels != 10 && levels != 20 && levels != 50 {
+			return fmt.Errorf("%w for `%s` only `1`, `5`, `10`, `20` or `50` is supported", subscription.ErrInvalidLevel, channel)
+		}
+	}
+	return nil
+}
+
 const subTplText = `
 {{- with $name := channelName $.S }}
 	{{- range $asset, $pairs := $.AssetPairs }}
 		{{- if singleSymbolChannel $name }}
 			{{- range $i, $p := $pairs -}}
+				{{- validateOrderbook $name $asset $.S.Interval $.S.Levels }}
```

Comment:
🚧  I propose something like [this](https://github.com/gbjk/gocryptotrader/commit/6d00d19fd5.patch)
```
curl https://github.com/gbjk/gocryptotrader/commit/6d00d19fd5.patch | git apply
```

Reply:
@gbjk THANKS SIRRRRRR!

Reply:
~Something like this~
[Here is the real juice](https://github.com/gbjk/gocryptotrader/commit/14b48f8c8f.patch) 🧃
* Tests are passing
* Fixed options orderbook generation
* Removed silent defaults from config, since we can't mix "params that get sent" with those that don't.
* I haven't yet updated docs; I'll do that in my PR upcoming
```
curl https://github.com/gbjk/gocryptotrader/commit/14b48f8c8f.patch | git apply
```
This is the complete patch, so can `git reset --hard HEAD^` or just apply it on top and it will probably jive okay.

This overlaps so heavily with my upcoming work that I don't want to go too far down the same route here on unifying.
I'll steal a bunch of this work when I rebase on top of this, and on top of the split PR.

Reply:
[2d6ebc7](https://github.com/thrasher-corp/gocryptotrader/pull/1863/commits/2d6ebc7ca81afb36f666f6da2537d61ed6bbbfd2)

Thanks!